FAQ.md

```diff
@@ -0,0 +1,112 @@
+-[@sdboyer in #371](https://github.com/golang/dep/issues/371#issuecomment-293246832)
+
+## Does `dep` replace `go get`?
+
+> I would say that dep doesn't replace go get, but they both can do similar things. Here's how I use them:
+>
+> `go get`: I want to download the source code for a go project so that I can work on it myself, or to install a tool. This clones the repo under GOPATH for all to use.
+>
+> `dep ensure`: I have imported a new dependency in my code and want to download the dependency so I can start using it. My workflow is "add the import to the code, and then run dep ensure so that the manifest/lock/vendor are updated". This clones the repo under my project's vendor directory, and remembers the revision used so that everyone who works on my project is guaranteed to be using the same version of dependencies.
+-[@carolynvs in #376](https://github.com/golang/dep/issues/376#issuecomment-293964655)
+
+> The long term vision is a sane, overall-consistent go tool. My general take is that `go get`
+> is for people consuming Go code, and dep-family commands are for people developing it.
+-[@sdboyer in #376](https://github.com/golang/dep/issues/376#issuecomment-294045873)
+
+## Why did `dep ensure -update` not update package X?
+
+
+* Is package X a direct dependency? [#385](https://github.com/golang/dep/issues/385)
+
+    Constraints given in a project's manifest are only applied if the
+    dependent project is actually imported. Transitive dependencies (dependencies
+    of your imports) are only updated when the revision in the lockfile no
+    longer meets the constraints of your direct dependencies.
+
+    > If you absolutely need to specify the constraint of a transitive dep from your own project, you have two options:
+    >
+    > Specify the constraint on github.com/gorilla/context via an override. Overrides apply globally, but are a power only given to the root project, so if anything else imports your project, the override won't be used.
+    > Mark github.com/gorilla/context as a required package in the manifest. This will cause it to be treated as a direct dependency, and your constraint will come into effect.
+    >
+    > However, before taking either of those steps, I'd say it's worth asking if you actually need to use master of github.com/gorilla/context. I imagine it's imported by github.com/gorilla/mux - and if that package is OK with using the tagged release instead of master (which is the preferred mode of operation anyway), then maybe that should be good enough for you? If you really needed something out of github.com/gorilla/context, then you'd probably be importing it directly and doing something with it
+    -[@sdboyer in #385](https://github.com/golang/dep/issues/385#issuecomment-294361087)
+
+* Is package X included in the `ignored` list in your manifest?
```

Comment:
If a package is ignored, then it shouldn't be showing up at all in lock or vendor - so there shouldn't be anything to update.